auto_round/export/export_to_awq.py

```diff
@@ -0,0 +1,253 @@
+    lr = kwargs["lr"]
+    minmax_lr = kwargs["minmax_lr"]
+    enable_minmax_tuning = kwargs["enable_minmax_tuning"]
+    enable_quanted_input = kwargs["enable_quanted_input"]
+    scale_dtype = kwargs["scale_dtype"]
+    tokenizer = kwargs["tokenizer"]
+    supported_types = kwargs["supported_types"]
+
+    logger.info("Saving quantized model to autoawq format")
+    if tokenizer is not None:
+        tokenizer.save_pretrained(output_dir)
+    ##check module quantized in block, this may have bug for mixed precision quantization
+    block_name = get_block_names(model)[0]
+    first_block = get_module(model, block_name)
+    all_to_quantized = True
+    modules_in_block_to_quantize = []
+    for n, m in first_block.named_modules():
+        is_supported_type = False
+        for supported_type in supported_types:
+            if isinstance(m, supported_type):
+                is_supported_type = True
+                break
+        if not is_supported_type:
+            continue
+        if not check_to_quantized(m):
+            all_to_quantized = False
+        else:
+            modules_in_block_to_quantize.append(n)
+    modules_in_block_to_quantize = [modules_in_block_to_quantize]
+    if all_to_quantized:
+        modules_in_block_to_quantize = None
+
+    compressed_model = copy.deepcopy(model.to("cpu"))
+
+    q_linear_module = WQLinear_GEMM
+    awq_model = AutoAWQForCausalLM.from_pretrained(model_path)
+    self_modules = awq_model.get_model_layers(compressed_model)
+    del awq_model  # release memory
+    for i in range(len(self_modules)):
+        module = self_modules[i]
+        named_linears = get_named_linears(module)
+        for name, linear_layer in named_linears.items():
+            key = get_module_name(compressed_model, linear_layer)
+            info = weight_config[key]
+            if not check_to_quantized(info):
+                continue
+            scale, zp = info["scale"], info["zp"]
+            scale = scale.t().contiguous()
+            zp = zp.t().contiguous()
+            q_linear = q_linear_module.from_linear(
+                linear=linear_layer,
+                w_bit=bits,
+                group_size=group_size,
+                init_only=False,
+                scales=scale,
+                zeros=zp,
+            )
+            linear_layer.cpu()
+            q_linear.to(next(module.parameters()).device)
+            set_op_by_name(module, name, q_linear)
+            clear_memory()
+
+    quant_config = {}
+    quant_config["quant_method"] = "awq"
+    quant_config["modules_to_not_convert"] = None
+    if compressed_model.config.model_type == 'mixtral':
+       quant_config["modules_to_not_convert"] = ["gate"] 
+    quant_config["version"] = "gemm"
+    quant_config["iters"] = iters
+    quant_config["lr"] = lr
+    quant_config["minmax_lr"] = minmax_lr
+    quant_config["enable_minmax_tuning"] = enable_minmax_tuning
+    quant_config["enable_quanted_input"] = enable_quanted_input
```

Comment:
follow autoround format to .serialize all the keys in the autoround.serialization_dict